### PR TITLE
[W5.11r][F12-B4]Jonathan Lee

### PR DESCRIPTION
**Pull request by jonleeyz**
LO-2KLoC

### CODE REVIEW

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -226,5 +282,7 @@ private Command prepareFind(String args) {
         return new FindCommand(keywordSet);
     }
 
-
+    public enum Attribute {
+        NAME, PHONE, EMAIL, ADDRESS
+    }
 }
```
Comment (OuyangDanwen) [inline src/seedu/addressbook/parser/Parser.java:288]: Same here. Try to remove the "no new line" character.

File: src/seedu/addressbook/parser/Parser.java [hunk 2]
```diff
@@ -156,6 +180,38 @@ private Command prepareDelete(String args) {
...
+        if (!matcher.matches()) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, EditCommand.MESSAGE_USAGE));
+        }
+        try {
+            String rawTargetIndex = matcher.group("targetIndex").trim();
+            String rawAttribute = matcher.group("attribute").trim();
+            final int targetIndex = parseArgsAsDisplayedIndex(rawTargetIndex);
+            final Attribute attribute = Attribute.valueOf(rawAttribute.toUpperCase());
+            final String newValue = matcher.group("newValue").trim();
+            return new EditCommand(
```
Comment (OuyangDanwen) [inline src/seedu/addressbook/parser/Parser.java:201]: No need to split here.

**Review by OuyangDanwen — COMMENTED** [review]
Good work! Just one thing missing to claim the credit. Try to write a few tests before the tutorial. @jonleeyz

**Comment by jonleeyz** [comment]
Did my best to write some new tests but can't seem to get the test for verifying successful edits working.

I have 2 problems specifically:

First, being the problem of easily and efficiently creating the reference Person objects that should be checked against for correctness.
Second, being the problem of extracting out the newly edited Person object after each EditCommand is executed and then being able to compare that object against the reference.

Would it be possible to obtain some clarity / help on these two issues?

Thank you!

**Pull request closed without merging**